packages/flame/lib/src/palette.dart

```diff
@@ -1,11 +1,26 @@
 import 'dart:ui';
 
+class _MemoizedPaint {
+  final Color color;
+  Paint? _paint;
+
+  _MemoizedPaint(this.color);
+
+  Paint get value {
+    _paint ??= Paint()..color;
+    return _paint!;
+  }
+}
```

Comment:
This does the same thing:

```suggestion
class _MemoizedPaint {
  final Color color;
  late final Paint value = Paint()..color = color;

  _MemoizedPaint(this.color);
}
```

Reply:
No it doesn't, your code already create an instance of the Paint, when the original code, only create the instance when it is accessed

Reply:
Yes, it does, `late` makes the field be lazy, everything after `=` is executed only when accessed for the first time.

Reply:
Oh, that is interesting, I thought late was just a way for the compiler to "trust" the programmer.

I will not apply this though, because we will probably don't even go with this solution